api/include/opentelemetry/nostd/list.h

```diff
@@ -0,0 +1,254 @@
+#pragma once
+
+#include <algorithm>
+#include <array>
+#include <cstring>
+#include <iostream>
+#include <iterator>
+#include <memory>
+
+#include "opentelemetry/nostd/utility.h"
+#include "opentelemetry/version.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace nostd
+{
+// Non-standard implementation of the list data structure.
+template <class T>
+class list
+{
+public:
+  typedef T value_type;
+
+  // Custom iterator class
+  class iterator
+  {
+  public:
+    iterator operator++()
+    {
+      ptr_++;
+      return ptr_;
+    }
+
+    T &operator*() { return *ptr_; }
+
+    T *operator->() { return ptr_; }
+
+    bool operator==(const iterator &other) { return ptr_ == other.ptr_; }
+
+    bool operator!=(const iterator &other) { return ptr_ != other.ptr_; }
+
+  private:
+    friend class list;
+
+    iterator(T *ptr) : ptr_(ptr) {}
+
+    T *ptr_;
+  };
+
+  list()
+  {
+    size_ = 0;
+    data_ = nullptr;
+  };
+
+  // Constructs class with passed in element.
+  list(T element)
+  {
+    size_ = 1;
+
+    data_ = new T[size_];
+
+    std::memcpy(data_, &element, sizeof(T));
```

Comment:
That's unsafe when used with smart pointers.

Reply:
Switched to std::copy.